### PR TITLE
Add flag to use FOR_BUILD.

### PR DESCRIPTION
**Pull request by jpakkane**
Needed for [this](https://salsa.debian.org/debian/debhelper/-/merge_requests/127#note_553685).

Tagged for 1.6.1. because this should go in the next Debian release.

*Edit*: do not merge until suitability has been agreed with Debian people.

### CODE REVIEW

**Comment by eli-schwartz** [comment]
Can the commit message be prefixed with "env2mfile: add [...]"? :)

**Comment by jpakkane** [comment]
Fixed.

File: mesonbuild/scripts/env2mfile.py [hunk 1]
```diff
@@ -421,12 +417,10 @@ def detect_missing_native_binaries(infos: MachineInfo) -> None:
             infos.binaries[toolname] = [exe]
 
 def detect_native_env(options: T.Any) -> MachineInfo:
-    use_for_build = has_for_build()
-    if use_for_build:
-        mlog.log('Using FOR_BUILD envvars for detection')
+    if options.use_for_build:
+        mlog.log('Using FOR_BUILD envvars for detection (native file for use during cross compilation)')
```
Comment (dnicolodi) [inline mesonbuild/scripts/env2mfile.py:421]: The help text uses `_FOR_BUILD`, this log message (and the commit message) use `FOR_BUILD`. I would use `_FOR_BUILD` everywhere, but maybe `_FOR_BUILD suffixed envvars` or `*_FOR_BUILD envvars` would be even better.

Comment (jpakkane) [reply mesonbuild/scripts/env2mfile.py]: Fixed, thanks.

**Pull request merged**